src/wallet/message_transporter.cpp

```diff
@@ -192,47 +192,48 @@ bool message_transporter::delete_message(const std::string &transport_id)
   return true;
 }
 
-// Deterministically derive a transport / Bitmessage address from 'seed' (the 10-hex-digits
-// auto-config token will be used), but do not set it up for receiving in PyBitmessage as
-// well, because it's possible the address will only ever be used to SEND auto-config data
+// Deterministically derive a new transport address from 'seed' (the 10-hex-digits auto-config
+// token will be used) and set it up for sending and receiving
+// In a first attempt a normal Bitmessage address was used here, but it turned out the
+// key exchange necessary to put it into service could take a long time or even did not
+// work out at all sometimes. Also there were problems when deleting those temporary
+// addresses again after auto-config. Now a chan is used which avoids all these drawbacks
+// quite nicely.
 std::string message_transporter::derive_transport_address(const std::string &seed)
 {
+  // Don't use the seed directly as chan name; that would be too dangerous, e.g. in the
+  // case of a PyBitmessage instance used by multiple unrelated people
+  const crypto::hash &hash = crypto::cn_fast_hash(seed.data(), seed.size());
```

Comment:
I believe this is taking a reference to a temporary. Though I have some vague recollection that modern C++ makes something similar OK, so that might be it ?

Reply:
Oh, all those possible pitfalls of C++.

With the use of `pod_to_hex` the reference will vanish and get replaced by a direct expression, so problem circumvented.